### DISCUSSION TITLE
Run dialog in another thread

### DISCUSSION THREAD

**Discussion by PontusLindberg** [PrismLibrary/Prism#2941] [WPF] [answered]
Hi!

I'm in the situation that I HAVE TO run a dialog in another thread. Is this possible to do within the Prism framework?

What I need to do more specifically is to show a progress dialog in another thread and not block the main thread (I can't do it the other way around and do the work in a background thread).

Kind regards,  Pontus

**Comment by brianlagunas** [comment, marked as answer]
Showing dialogs in separate threads is not the responsibility of Prism. You can easily create your own multi-threaded dialog service and register it with Prism, but you're responsible for creating the multi-threaded dialog.

**Reply by PontusLindberg** [reply]
Ok, get it, thanks for a quick answer, very much appreciated. Do you know of something similar done before that I can take a look at? 

Kind regards, Pontus

**Reply by brianlagunas** [reply]
No, in my 20 years of experience, I've never had to open a dialog in a separate UI thread. Happy coding 😄
